src/lib/BLE/devBLETelemetry.cpp

```diff
@@ -0,0 +1,301 @@
+    p->channels.ch12 = ChannelData[12];
+    p->channels.ch13 = ChannelData[13];
+    p->channels.ch14 = ChannelData[14];
+    p->channels.ch15 = ChannelData[15];
+
+    outBuffer[CRSF_TELEMETRY_TYPE_INDEX + 1 + RCframeLength] = crsf_crc.calc((byte *)&outBuffer[CRSF_TELEMETRY_TYPE_INDEX], RCframeLength + 1 );
+
+    rcCRSF->setValue(outBuffer, RCframeLength + 4);
+    rcCRSF->notify();
+}
+
+void BluetoothTelemetryShutdown()
+{
+    if ( pServer != nullptr)
+    {
+        DBGLN("Stopping BLE Telemetry!");
+
+        NimBLEDevice::stopAdvertising();
+        NimBLEDevice::deinit(true);
+        pServer = nullptr;
+        rcCRSF = nullptr;
+        vSemaphoreDelete(_mutex);
+    }
+}
+
+static void BluetoothTelemetryUpdateDevice()
+{
+    if ( (config.GetBLETelemetry() == false) || (connectionState == bleJoystick) )
+    {
+        BluetoothTelemetryShutdown();
+        return;
+    }
+
+    // pServer is null if it hasn't been started yet
+    if (pServer != nullptr)
+        return;
+
+    _mutex = xSemaphoreCreateMutex();
+
+    //NimBLEDevice::init(String(String(device_name) + " " + getMasterUIDString()).c_str());
+    NimBLEDevice::init("ExpressLRS Telemetry");
+
+    //we do not want devices which are bound to BLE Joystick to connect to Telemetry service.
+    //start BLE Device with random address
+    //avoid frequent address changes
+    if ( initOnce == false )
+    {
+        initOnce = true;
+
+        ble_addr_t blead;
+        ble_hs_id_gen_rnd(1, &blead);
+        ble_hs_id_set_rnd(blead.val);
+    }
+    NimBLEDevice::setOwnAddrType(BLE_OWN_ADDR_RANDOM);
+
+    //Set MTU to max packet length + 3 bytes to be able to send packets longer then default 20 bytes.
+    //Should be set on both ends - smaller from two is used.
+    BLEDevice::setMTU(CRSF_MAX_PACKET_LEN + 3);
+
+    /** Set low transmit power, default is 6db */
+    BLEDevice::setPower(ESP_PWR_LVL_P6);
+    pServer = NimBLEDevice::createServer();
+    NimBLEService *rcService = pServer->createService(TELEMETRY_SVC_UUID);
+    rcCRSF = rcService->createCharacteristic(TELEMETRY_CRSF_UUID, NIMBLE_PROPERTY::READ | NIMBLE_PROPERTY::NOTIFY, CRSF_MAX_PACKET_LEN);
+    rcService->start();
+
+    auto dInfo = pServer->createService(DEVICE_INFO_SVC_UUID);
+    dInfo
+        ->createCharacteristic(MANUFACTURER_NAME_SVC_UUID, NIMBLE_PROPERTY::READ)
+        ->setValue("ExpressLRS");
+    dInfo
+        ->createCharacteristic(MODEL_NUMBER_SVC_UUID, NIMBLE_PROPERTY::READ)
+        ->setValue(String(product_name));
+    dInfo
+        ->createCharacteristic(SERIAL_NUMBER_SVC_UUID, NIMBLE_PROPERTY::READ)
+        ->setValue(String(__TIMESTAMP__) + " - " + getMasterUIDString());
+    dInfo
+        ->createCharacteristic(SOFTWARE_NUMBER_SVC_UUID, NIMBLE_PROPERTY::READ)
+        ->setValue(String(version));
+    dInfo
+        ->createCharacteristic(HARDWARE_NUMBER_SVC_UUID, NIMBLE_PROPERTY::READ)
+        ->setValue(String(getRegulatoryDomain()));
+
+    dInfo->start();
```

Comment:
```suggestion
    dInfo->start();
    
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_DEFAULT);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_ADV);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL0);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL1);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL2);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL3);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL4);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL5);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL6);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL7);
    NimBLEDevice::setPower(ESP_PWR_LVL_P9, ESP_BLE_PWR_TYPE_CONN_HDL8);    
```

Call at correct place, also use maximum power to ease life of users with poor antennas badly installed inside TX modules...

Reply:
shouldnt setting only ESP_BLE_PWR_TYPE_DEFAULT be enough? According to the docu default   is used for all other modes that were not set explicitly

Reply:
Setting default only does not have effect.
I checked RSSI with this app:
https://play.google.com/store/apps/details?id=com.macdom.ble.blescanner&hl=en

RSSI improves if all settings are set.

Reply:
power level set to 4mW instead